pp/go/relabeler/head/wal_writer.go

```diff
@@ -75,13 +103,26 @@ func (w *segmentWriter) sync() error {
 }
 
 func (w *segmentWriter) encodeAndFlush(segment EncodedSegment) (encoded bool, err error) {
-	if _, err := WriteSegment(w.buffer, segment); err != nil {
+	n, err := WriteSegment(w.buffer, segment)
+	if err != nil {
 		w.buffer.Reset()
 		return false, fmt.Errorf("encode segment: %w", err)
 	}
 
 	w.writeCompleted = false
-	return true, w.flushAndSync()
+
+	if err := w.flushAndSync(); err != nil {
+		return true, err
+	}
+
+	if w.initSize != 0 {
+		w.walSize.Set(float64(w.initSize))
+		w.initSize = 0
+	}
+
+	w.walSize.Add(float64(n))
```

Comment:
pass this gauge to writer or [wrap registerer with label shard_id](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#WrapRegistererWith) and write actual offset after write. Here metric value may be incorrect